src/components/structures/RoomView.js

```diff
@@ -785,29 +795,57 @@ export default createReactClass({
             return;
         }
         const e2eMembers = await room.getEncryptionTargetMembers();
+
+        /*
+        Ensure we trust our own signing key, ie, nobody's used our credentials to
+        replace it and sign all our devices
+        */
+        if (!cli.checkUserTrust(cli.getUserId())) {
+            this.setState({
+                e2eStatus: "warning",
+            });
+            debuglog("e2e status set to warning due to not trusting our own signing key");
+            return;
+        }
+
+        /*
+        Gather verification state of every user in the room.
+        If _any_ user is verified then _every_ user must be verified, or we'll bail.
+        Note we don't count our own user so that the all/any check behaves properly.
+        */
+        const verificationState = e2eMembers.map(({userId}) => userId)
+            .filter((userId) => userId !== cli.getUserId())
+            .map((userId) => cli.checkUserTrust(userId).isCrossSigningVerified());
+        if (verificationState.includes(true) && verificationState.includes(false)) {
+            this.setState({
+                e2eStatus: "warning",
```

Comment:
This isn't quite what we want: we need to do the `allDevicesVerified` stuff first, since the room should only be `warning` if there is a user for which `checkUserTrust(u).isVerified() && ! allDevicesVerified`. Otherwise it should be `normal`.

Reply:
oh, good point

Reply:
fixèd